app/src/full/kotlin/io/homeassistant/companion/android/data/wear/WearDataListener.kt

```diff
@@ -0,0 +1,36 @@
+package io.homeassistant.companion.android.data.wear
+
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.wearable.WearableListenerService
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.tasks.asTask
+import okhttp3.Dns
+
+@AndroidEntryPoint
+class WearDataListener : WearableListenerService() {
+
+    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    override fun onRequest(nodeId: String, path: String, request: ByteArray): Task<ByteArray>? {
+        if (path == "/dnsLookup") {
+            val hostname = String(request, Charsets.UTF_8)
+            return mainScope.async(Dispatchers.IO) {
+                Dns.SYSTEM.lookup(hostname).first().address
```

Comment:
Why only returning the first? If I'm not wrong you could get at two InnetAddress one for ipv4 and one for ipv6. The wear could benefits having the two.

It avoid having some logic here you just forward whatever you received without transformation.

Reply:
Yep. Should I json encode the request and response? Would be more inline with other uses?

This actually reminds me, we should prefer IPv4, as I'm not sure the BT proxied connection supports IPv6.

Reply:
I'm ok with JSON. Even if we should prefer IPv4 let's not take any decision at this level. The wear should deal with that not the mobile.

Reply:
Went csv